### ISSUE TITLE
investigate Toml big string tests 

### ISSUE THREAD

**Issue by pjfanning** [FasterXML/jackson-dataformats-text#384]
* tests added in #383
* testBiggerString indicates that only the TextBuffer.contentAsString() check is catching the issue with the string being too big
  * no matter how big the len is set to
  * this is far from ideal - we should be catching exceptions when the buffer is extended but the toml code has its own logic and somehow bypasses the checks in TextBuffer until the contentAsString() call.

**Comment by cowtowncoder** [comment]
If I remember correctly, with TOML content is first parsed as something else and only then processed further. I think it generates and uses a Lexer to do that, and that would do its own handling.

I think @yawkat knows more about this and could perhaps help.

**Comment by yawkat** [comment]
@pjfanning toml only does TextBuffer.append in its Lexer, it is basically used like a StringBuilder. the buffer is created in the generated Lexer class using constructTextBuffer, so this may be part of the problem. I also checked the TextBuffer implementation, and it looks like TextBuffer does not check the length on append?

I tried running the tests you pushed, but they fail already, the check in contentAsString is not triggered like you say. So I'm not sure if I'm misunderstanding it.
